charms/dex-auth/reactive/dex_auth.py

```diff
@@ -19,7 +19,9 @@ def charm_ready():
     layer.status.active("")
 
 
-@when_any("layer.docker-resource.oci-image.changed", "config.changed")
+@when_any(
+    "layer.docker-resource.oci-image.changed", "config.changed", "endpoint.oidc-client.changed"
```

Comment:
The `config.changed` flag gets cleared automatically for legacy reasons, but I don't think that's necessarily true of the other two.  Have you confirmed that this doesn't lead to a "restart on every `update-status` hook" issue?